### PR TITLE
Fix `cargo kani --debug` and remove debug logs from `--verbose`

### LINKED ISSUE #1348: --debug doesn't work anymore
```
$ cargo kani --debug --verbose
KANIFLAGS="--goto-c --log-level=debug --assertion-reach-checks -C symbol-mangling-version=v0" RUSTC="/home/ubuntu/rmc/target/debug/kani-compiler" RUSTFLAGS="--kani-flags" cargo build --target x86_64-unknown-linux-gnu --target-dir target -v
error: output of --print=file-names has changed in the compiler, cannot parse
command was: `/home/ubuntu/rmc/target/debug/kani-compiler - --crate-name ___ --print=file-names --kani-flags --target x86_64-unknown-linux-gnu --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg`

--- stdout
DEBUG kani_compiler Sysroot path., path="/home/ubuntu/.rustup/toolchains/nightly-2022-06-09-x86_64-unknown-linux-gnu", sysroot_arg=None
DEBUG kani_compiler Compile, rustc_args=["rustc", "-C", "overflow-checks=on", "-C", "panic=abort", "-Z", "unstable-options", "-Z", "panic_abort_tests=yes", "-Z", "trim-diagnostic-paths=no", "-Z", "human_readable_cgu_names", "--cfg=kani", "-Z", "crate-attr=feature(register_tool)", "-Z", "crate-attr=register_tool(kanitool)", "-L", "/home/ubuntu/rmc/target/debug/build/kani-compiler-28743720f584101a/out/lib", "--extern", "kani", "--extern", "noprelude:std=/home/ubuntu/rmc/target/debug/build/kani-compiler-28743720f584101a/out/lib/libstd.rlib", "-C", "symbol-mangling-version=v0", "-", "--crate-name", "___", "--print=file-names", "--target", "x86_64-unknown-linux-gnu", "--crate-type", "bin", "--crate-type", "rlib", "--crate-type", "dylib", "--crate-type", "cdylib", "--crate-type", "staticlib", "--crate-type", "proc-macro", "--print=sysroot", "--print=cfg", "--sysroot", "/home/ubuntu/.rustup/toolchains/nightly-2022-06-09-x86_64-unknown-linux-gnu"]
___
lib___.rlib
lib___.so
lib___.so
lib___.a
lib___.so
/home/ubuntu/.rustup/toolchains/nightly-2022-06-09-x86_64-unknown-linux-gnu
debug_assertions
kani
panic="abort"
proc_macro
target_abi=""
target_arch="x86_64"
target_endian="little"
target_env="gnu"
target_family="unix"
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="64"
target_has_atomic="8"
target_has_atomic="ptr"
target_has_atomic_equal_alignment="16"
target_has_atomic_equal_alignment="32"
target_has_atomic_equal_alignment="64"
target_has_atomic_equal_alignment="8"
target_has_atomic_equal_alignment="ptr"
target_has_atomic_load_store="16"
target_has_atomic_load_store="32"
target_has_atomic_load_store="64"
target_has_atomic_load_store="8"
target_has_atomic_load_store="ptr"
target_os="linux"
target_pointer_width="64"
target_thread_local
target_vendor="unknown"
unix

Error: cargo exited with status exit status: 101
```

### LINKED ISSUE #1631: --verbose doesn't work anymore
Using `--verbose` causes obscene levels of debug spam now, almost like `--debug` was turned on.

It's no longer useful for trying to get a little bit more output from subcommands.

Example:

```
┐kani_compiler::codegen_cprover_gotoc::codegen::function::CodegenFunction name="delegate_use_struct"
┐kani_compiler::codegen_cprover_gotoc::codegen::function::CodegenFunction name="delegate_use_struct"
├─┐
├───┐kani_compiler::codegen_cprover_gotoc::codegen::statement::CodegenTerminator statement=_1 = dependency3::give_foo() -> bb1
├─┐
├───┐kani_compiler::codegen_cprover_gotoc::codegen::statement::CodegenTerminator statement=_1 = dependency3::give_foo() -> bb1
│ ┌─┘
├─┘
│ ┌─┘
├─┘
├─┐
```

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

- Fix #1348: Fix `cargo kani --debug` by redirecting kani-compiler logs to the STDERR so it doesn't conflict with cargo's output expectations.
- Fix #1631: Remove `kani-compiler` logs from the output of `--verbose`.

### Resolved issues:

Fix #1631
Fix #1348

### Related RFC:

N/A

### Call-outs:

These were tiny fixes so I combined them in one PR.

### Testing:

* How is this change tested? N/A

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-driver/src/call_single_file.rs [hunk 1]
```diff
@@ -107,8 +107,6 @@ impl KaniSession {
 
         if self.args.debug {
             flags.push("--log-level=debug".into());
-        } else if self.args.verbose {
-            flags.push("--log-level=info".into());
```
Comment (tedinski) [inline kani-driver/src/call_single_file.rs:111]: Was this behavior undesirable? Not sure why this was changed.

The stderr logging was enough to fix, right?

**Review by tedinski — APPROVED** [review]
probably fine, but I had a question

File: kani-driver/src/call_single_file.rs (hunk 1 above)
Comment (celinval) [reply kani-driver/src/call_single_file.rs:111]: This was to fix #1631

Comment (tedinski) [reply kani-driver/src/call_single_file.rs:111]: Hah, despite me being the one to file those issues, I'd forgotten what the verbose one was actually about.

Makes sense! :)

The CI failure is a mystery though...

Comment (celinval) [reply kani-driver/src/call_single_file.rs:111]: I found out the issue. Submitting a patch now.

**Review by tedinski — APPROVED** [review]

**Pull request merged**